[T16-1] Trajectory

m-aslam-mj2 opened the pull request.

stephlewyh commented on docs/DeveloperGuide.adoc:
```diff
@@ -317,7 +316,238 @@ image::UndoRedoActivityDiagram.png[width="650"]
+
+===== Aspect: How graph-to-data executes
+
+* **Alternative 1 (current choice):** Individual command knows how to export accordingly.
```
You may consider rewording this. 

A command 'does not know' how to export. Executing the command results in (Something) etc.

stephlewyh commented on docs/DeveloperGuide.adoc:
```diff
@@ -317,7 +316,238 @@ image::UndoRedoActivityDiagram.png[width="650"]
+
+The following activity diagram summarizes what happens when a user executes a new command:
+
+image::dataToGraphActivityDiagram.png[width="650"]
```
Again, good job here in illustrating your point with an activity diagram.

stephlewyh reviewed (commented):
Dear team T16-1, your team is progressing well. Some points to note regarding documentation: 
Generally the 9 features presented in your developer guide well-defined in terms of scope. 

However, I only noticed 2 new activity diagrams and no other UML diagram. You may want to consider using other diagrams to illustrate your point. As a rough guide, there should be one UML diagram per head.  Also, you should draw relevant parts, not low-level details across multiple components / big diagrams.

jomcruz93 committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T16-1/main

merge upstream

zhicaizack committed:
Merge pull request #257 from jomcruz93/master

Fixed a bug in note edit command

jomcruz93 committed:
Improved error messages when entering invalid inputs for note add command

jomcruz93 committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T16-1/main

Merged with upstream

m-aslam-mj2 committed:
Merge pull request #284 from jomcruz93/master

Corrected error message when entering invalid parameters for note add command

jomcruz93 committed:
Added validation to day of month.
Refactored some codes.

zhicaizack committed:
Update UG to fix error in documentation

jomcruz93 committed:
Merge pull request #286 from CS2113-AY1819S1-T16-1/zhicaizack-patch-1

Update UG to fix error in documentation

jomcruz93 committed:
Removed unnecessary lines of code and updated JavaDoc comment correspondingly in NoteTime class

jomcruz93 committed:
Removed unnecessary lines of code in NoteDate class.
Updated JavaDoc comment corresponding to the changes.

jomcruz93 committed:
Added a feature to auto-complete command words with TAB.

harriuscai committed:
Merge pull request #285 from jomcruz93/master

Added validation for day of month

jomcruz93 committed:
Enhanced auto-complete feature: Cycle between matching keywords is now allowed.

jomcruz93 committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T16-1/main into feature-auto-complete-cli

pulled latest code from upstream

jomcruz93 committed:
Slightly modified the codes for autocomplete function

jomcruz93 committed:
Set auto-complete function only for the first two words.

zhicaizack committed:
* Added validation to whether student is enrolled in module before assigning student to classroom

m-aslam-mj2 committed:
Swapped order of features in UG, removed access to undo, redo and clear commands.

Signed-off-by: m-aslam-mj2 <e0176811@u.nus.edu>

zhicaizack committed:
* Introduced the draft documentation of the PPP
* Tagged the documentation

zhicaizack committed:
* Fixed typo in template

tristyxxnana committed:
tristyxxnana.adoc

zhicaizack committed:
* Added newline at EOF

harriuscai committed:
Merge pull request #290 from zhicaizack/ppp

Introduced the draft documentation of the PPP

tristyxxnana committed:
Merge branch 'updatedUserDeveloperGuide' into PPP

# Conflicts:
#	docs/DeveloperGuide.adoc
#	docs/UserGuide.adoc

tristyxxnana committed:
updated tristyxxnana.adoc

harriuscai committed:
Merge pull request #291 from m-aslam-mj2/stage

Rectified issues found in PE1

jomcruz93 committed:
Updated UG and DG. Added PPP.

jomcruz93 committed:
Resolved merge conflict.

tristyxxnana committed:
updated according to PR feedback

jomcruz93 committed:
Fixed a typo in DG.

harriuscai committed:
Merge pull request #292 from tristyxxnana/PPP

Project Portfolio Page

harriuscai committed:
Merge pull request #293 from jomcruz93/master

Added my PPP

harriuscai committed:
Add the first draft of the Project Portfolio Page for harriuscai

harriuscai committed:
Add newline at EOF

m-aslam-mj2 committed:
Merge pull request #294 from harriuscai/master

Add the first draft of the Project Portfolio Page for harriuscai

m-aslam-mj2 committed:
First draft of PPP.

Signed-off-by: m-aslam-mj2 <e0176811@u.nus.edu>

m-aslam-mj2 committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T16-1/main

m-aslam-mj2 committed:
Update to first draft of PPP.

Signed-off-by: m-aslam-mj2 <e0176811@u.nus.edu>

harriuscai committed:
Merge pull request #295 from m-aslam-mj2/stage8

Added my first draft of the PPP.

tristyxxnana committed:
updated grade add to refresh list instantly

tristyxxnana committed:
Merge branch 'PPP' into updateList

jomcruz93 committed:
Enhanced note delete command to accept range of indexes
Updated test codes accordingly
Corrected other test cases for note management

jomcruz93 committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T16-1/main

merged latest upstream with local repo

jomcruz93 committed:
Updated test codes

jomcruz93 committed:
Rewritten some codes in NoteDeleteCommandParser

tristyxxnana committed:
tests

harriuscai committed:
Merge pull request #297 from jomcruz93/master

Enhancement to Note Delete Command

jomcruz93 committed:
Adjusted codes to allow duplicate indexes in note delete

zhicaizack committed:
* Added validation to remove from class delete student command

m-aslam-mj2 committed:
PPP now contains link to project dashboard.

jomcruz93 committed:
Fixed checkstyle error

harriuscai committed:
Fix some formatting issues in the User Guide

harriuscai committed:
Merge pull request #298 from zhicaizack/validatemoduleenrol

Validates whether a student is enrolled into a module before assigning the student to classroom

harriuscai committed:
Update config.json

harriuscai committed:
Merge pull request #299 from jomcruz93/master

Removed validation for duplicate indexes for faster execution

zhicaizack committed:
* Fix formatting error in userguide and developer guide

zhicaizack committed:
* Update the UI Component of developer guide

tristyxxnana committed:
updated doc with contribution link

m-aslam-mj2 committed:
Merge pull request #300 from harriuscai/master

Fix some formatting issues in the User Guide

m-aslam-mj2 committed:
Merge pull request #301 from zhicaizack/userdevguidefix

Fix documentation error in user guide and developer guide

m-aslam-mj2 committed:
Merge pull request #302 from zhicaizack/UIUpdate

Update the UI compoenent of developer guide

m-aslam-mj2 committed:
Pre-typed commands, that can be navigated between with right/left arrow. For presentation use.

m-aslam-mj2 committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T16-1/main

jomcruz93 committed:
Merge branch 'master' into feature-auto-complete-cli

merge with master branch

m-aslam-mj2 committed:
Merge pull request #303 from tristyxxnana/updatedPPP

Updated PPP

m-aslam-mj2 committed:
Added splash screen, added tests.

m-aslam-mj2 committed:
Update .gitignore

m-aslam-mj2 committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T16-1/main

jomcruz93 committed:
Enhanced auto-complete feature

m-aslam-mj2 committed:
Added 'debug' command for populating sample data. Moved inputs.txt outside of data folder.

m-aslam-mj2 committed:
Fixed EOF error.

jomcruz93 committed:
Enhanced auto-complete function

jomcruz93 committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T16-1/main into feature-auto-complete-cli

merged with upstream

jomcruz93 committed:
Fixed checkstyle violations in tests

harriuscai committed:
Merge pull request #304 from m-aslam-mj2/stage10

Added tests, added splash screen, added pre-typed commands.

jomcruz93 committed:
Merge branch 'master' into feature-auto-complete-cli

merged with local master branch

tristyxxnana committed:
Merge branch 'updatedPPP' into v1.4

# Conflicts:
#	docs/team/tristyxxnana.adoc

tristyxxnana committed:
Merge branch 'master' into v1.4

jomcruz93 committed:
Re-arranged keywords in CliSyntax in alphabetical order

jomcruz93 committed:
Minor changes to auto-complete method

tristyxxnana committed:
Updated logic diagram and removed failed test methods

tristyxxnana committed:
removed unnecessay imports

tristyxxnana committed:
removed docs

tristyxxnana committed:
removed docs related changes

tristyxxnana committed:
added diagram pic to image folder

tristyxxnana committed:
updated logic component

tristyxxnana committed:
updated documentation with appropriate links

m-aslam-mj2 committed:
Merge pull request #312 from tristyxxnana/v1.4test

Added test methods

m-aslam-mj2 committed:
Merge pull request #311 from jomcruz93/master

Fixed checkstyle violations in tests

jomcruz93 committed:
Added new class diagrams for Model component

tristyxxnana committed:
updated DG manual testing section

tristyxxnana committed:
Merge branch 'v1.4logicdiagram' into v1.4documentation

# Conflicts:
#	docs/UserGuide.adoc
#	docs/team/tristyxxnana.adoc

tristyxxnana committed:
made changes to errors

tristyxxnana committed:
updated arrows to touch the block

harriuscai committed:
Merge pull request #316 from tristyxxnana/v1.4documentation

Updated Documentation

zhicaizack committed:
* WIP
* Added tests for CRUD of Classroom-releated commands
* Fixed typo in variable naming

zhicaizack committed:
* Fix missing javadoc comment for checkstyles for test related class

zhicaizack committed:
* Added teardown after test cases
* Fix testcase failure locally

jomcruz93 committed:
Updated Model class diagram

jomcruz93 committed:
Made MODULE_CODE param optional when adding notes.
Improved validation when reading from XML data.
Changed character limit of notes from 150 to 300.
Changed the way note title is presented in CSV format

jomcruz93 committed:
Updated relevant test codes accordingly to work on the new changes

jomcruz93 committed:
Fixed test failures in NoteManagerTest class

jomcruz93 committed:
Corrected a line of code in XmlAdaptedNote toModelType() method

jomcruz93 committed:
Minor changes in NoteManager class

harriuscai committed:
Merge pull request #315 from jomcruz93/docs-diagrams

Added new class diagram files for Model component

zhicaizack committed:
* Shifted methods from `ClassroomParserUtil` to `ParserUtil`
* Added some valid/invalid classroom details to `CommandTestUtil`
* Added tests for classroom-related CommandParsers

jomcruz93 committed:
Changed the background of ModelClassDiagram.png to be transparent.
Updated Model component description in DG.

tristyxxnana committed:
added new activity diagram to implementation section

tristyxxnana committed:
added explanation

tristyxxnana committed:
changed format

tristyxxnana committed:
Merge pull request #320 from jomcruz93/docs-diagrams

Updated Model component description in DG

jomcruz93 committed:
Added Instructions for Manual Testing for Notes feature

jomcruz93 committed:
Slight modification to Instructions for manual testing

zhicaizack committed:
* Added some tests for class assigning/unassigning of students
* Test fails when running all the tests together, failed tests are commented.

m-aslam-mj2 committed:
Merge pull request #319 from jomcruz93/module-code-as-optional-input

Set MODULE_CODE as optional parameter when creating notes..

m-aslam-mj2 committed:
Merge pull request #322 from jomcruz93/master

Added Instructions for Manual Testing for Notes feature in DG

m-aslam-mj2 committed:
Merge pull request #321 from tristyxxnana/v1.4implementation

Replaced activity diagram in implementation section

jomcruz93 committed:
Merge branch 'module-code-as-optional-input'

merged branch as it was already merge to upstream

jomcruz93 committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T16-1/main

merged with latest from upstream

jomcruz93 committed:
Removed NoteManager#initNoteManager(). Updated tests accordingly

jomcruz93 committed:
Fixed checkstyle violation

zhicaizack committed:
* Fix testcase failure by using another method of enrolling students into a module.

m-aslam-mj2 committed:
Merge pull request #318 from zhicaizack/testfiles

Add testcase for classroom features

m-aslam-mj2 committed:
Merge pull request #323 from jomcruz93/master

Removed redundant class in NoteManager

zhicaizack committed:
* Global fix for import having wrong orders, some files having more than 120 characters.

zhicaizack committed:
* Fix import for `TypicalPersons`

jomcruz93 committed:
Changed directory of exported CSV files.
Refactored codes for NoteExportCommand.
Updated tests for NoteExportCommand.

m-aslam-mj2 committed:
Merge pull request #325 from zhicaizack/globalCSfix

Added javadoc comments for most test files

m-aslam-mj2 committed:
Merge pull request #326 from jomcruz93/update-export-directory

Changed directory of exported CSV files

zhicaizack committed:
Merge branch 'feature-auto-complete-cli' of https://github.com/jomcruz93/main

jomcruz93 committed:
Added more test cases

jomcruz93 committed:
Merged with upstream. Fixed merge conflicts

tristyxxnana committed:
resolved checkstyle errors

zhicaizack committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T16-1/main

jomcruz93 committed:
Added tearDown() for CsvUtilTest

jomcruz93 committed:
Minor change to NoteEditCommandTest

harriuscai committed:
Merge branch 'master' into feature-auto-complete-cli

harriuscai committed:
Merge pull request #313 from jomcruz93/feature-auto-complete-cli

Auto-complete commands with TAB

harriuscai committed:
Modify the architecture diagram, and add a new sequence diagram to illustrate the MVC implementation

harriuscai committed:
Add a write-up in the architecture design section to describe the MVC implementation in Trajectory.

harriuscai committed:
Modify a link in the logic design section to use the repoUrl variable

m-aslam-mj2 committed:
Added lock/unlock capability to Trajectory. Expedited commands now require CTRL modifier key.

Signed-off-by: m-aslam-mj2 <e0176811@u.nus.edu>

m-aslam-mj2 committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T16-1/main

m-aslam-mj2 committed:
Merge pull request #327 from jomcruz93/added-tests

Added more test cases

m-aslam-mj2 committed:
Merge pull request #328 from harriuscai/master

Completed the write-up on the Architecture design section of the DeveloperGuide

m-aslam-mj2 committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T16-1/main

harriuscai committed:
Merge pull request #330 from m-aslam-mj2/stage11

Added lock/unlock capability. Expedited commands now require modifier key.

zhicaizack committed:
* Added test files for the rest of the class related commands
* Removed some redundant code from test files

zhicaizack committed:
* Fix logic error for test case for classroom attendance

zhicaizack committed:
Merge branch 'master' of https://github.com/zhicaizack/main

m-aslam-mj2 committed:
Updated DG. Added 'course edit' command and fixed a bug with 'student edit'.

Signed-off-by: m-aslam-mj2 <e0176811@u.nus.edu>

m-aslam-mj2 committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T16-1/main

m-aslam-mj2 committed:
Merge pull request #331 from zhicaizack/testfiles

Added more test cases for class related features

zhicaizack committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T16-1/main

harriuscai committed:
Merge pull request #332 from m-aslam-mj2/stage12

 Updated DG. Added 'course edit' command and fixed a bug with 'student edit'.

jomcruz93 committed:
Added sequence diagram for note export implementation

jomcruz93 committed:
Adjusted size of diagram in DG

jomcruz93 committed:
Adjusted size of diagram in DG 2

jomcruz93 committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T16-1/main into docs

merge with upstream

jomcruz93 committed:
Fixed minor error in DG

harriuscai committed:
Add UI support for module list and find

harriuscai committed:
Fix a bug that allowed more than one module with the same module code to be added

harriuscai committed:
Add newline at EOF

harriuscai committed:
Improve the summary message when enrolling students in a module

jomcruz93 committed:
Added validation if module exists when adding notes to module.
Updated tests accordingly.

m-aslam-mj2 committed:
added a few tests.

Signed-off-by: m-aslam-mj2 <e0176811@u.nus.edu>

m-aslam-mj2 committed:
Merge pull request #333 from jomcruz93/docs

Updated UG, DG, and PPP

m-aslam-mj2 committed:
Merge pull request #334 from harriuscai/master

Add UI support for module list and find

m-aslam-mj2 committed:
Merge pull request #335 from jomcruz93/link-note-with-modulecode

Added validation if module exists when adding notes to module.

m-aslam-mj2 committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T16-1/main

zhicaizack committed:
* Updated DG for manual test section
* Rearranged the order of class features in UG and DG
* Updated PPP for corresponding change

harriuscai committed:
Create utility methods to help display details of an entity such as module

harriuscai committed:
Add UI support for module view

harriuscai committed:
Add UI support for module enrol to display the details of the module after enrolment

harriuscai committed:
Add UI support for module add, delete, and edit

harriuscai committed:
Add newline at EOF

harriuscai committed:
Merge pull request #336 from m-aslam-mj2/staging13

Added a few tests.

harriuscai committed:
Merge pull request #337 from zhicaizack/docupdate

Docupdate for Manual Test, Fixes in UG and DG

harriuscai committed:
Merge pull request #338 from harriuscai/master

Add UI support for module view, enrol, add, delete, and edit

harriuscai committed:
Add sample module data to the debug command

zhicaizack committed:
* Changed display of class related info to banner type

harriuscai committed:
Sort modules by module code in alphabetical order

harriuscai committed:
Add newline at EOF

harriuscai committed:
Merge pull request #339 from zhicaizack/ui

Changed display of class related info to banner type

m-aslam-mj2 committed:
Updated UG and DG.

Signed-off-by: m-aslam-mj2 <e0176811@u.nus.edu>

m-aslam-mj2 committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T16-1/main

m-aslam-mj2 committed:
Merge pull request #340 from harriuscai/master

Add sample data in DebugCommand

m-aslam-mj2 committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T16-1/main

m-aslam-mj2 committed:
Merge pull request #341 from m-aslam-mj2/staging14

Updated UG and DG.

harriuscai committed:
Fix some checkstyle errors

harriuscai committed:
Merge pull request #342 from harriuscai/master

Fix some checkstyle errors

tristyxxnana committed:
test files added

tristyxxnana committed:
Merge branch 'v1.4testing' into testingfiles

tristyxxnana committed:
Update suppressions.xml

zhicaizack committed:
* Added basic sequence diagram for classroom add

m-aslam-mj2 committed:
Merge pull request #343 from zhicaizack/umldoc

Added uml diagram into DG

jomcruz93 committed:
Updated the rendering of HTML cards for notes

jomcruz93 committed:
Removed a line of code from NoteListCommand

m-aslam-mj2 committed:
Merge pull request #344 from tristyxxnana/testingfiles

Test Files Added for Grade and Gradebook

m-aslam-mj2 committed:
fixed checkstyle errors, added some tests, updated PPP.

Signed-off-by: m-aslam-mj2 <e0176811@u.nus.edu>

m-aslam-mj2 committed:
Merge pull request #345 from jomcruz93/update-html-card

Updated the rendering of HTML cards for notes

m-aslam-mj2 committed:
added DG-related data.

Signed-off-by: m-aslam-mj2 <e0176811@u.nus.edu>

m-aslam-mj2 committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T16-1/main

m-aslam-mj2 committed:
removed files that were meant to be versioned.

Signed-off-by: m-aslam-mj2 <e0176811@u.nus.edu>

tristyxxnana committed:
removed links

tristyxxnana committed:
removed links

tristyxxnana committed:
updated DG

m-aslam-mj2 committed:
Merge pull request #346 from m-aslam-mj2/staging15

Fixed CS errors, added some tests, updated PPP.

zhicaizack committed:
* Populated `DebugCommand` with sample datas

m-aslam-mj2 committed:
Update asciidoctor.css

m-aslam-mj2 committed:
Merge pull request #347 from zhicaizack/master

Populate sample data

zhicaizack committed:
* Shorten PPP
* Fix uml diagram not showing up on DG

zhicaizack committed:
* Renamed class add sequence diagram

m-aslam-mj2 committed:
Updated student list view. Updated UG and DG. Updated build.gradle.

Signed-off-by: m-aslam-mj2 <e0176811@u.nus.edu>

zhicaizack committed:
* Shifted class add sequence diagram to correct order
* Renamed file so adoc can see the picture

m-aslam-mj2 committed:
Merge pull request #348 from zhicaizack/docupdatev2

Fix visual bug on DG and shorten PPP

zhicaizack committed:
* Updated manual testing to use sample module and student

tristyxxnana committed:
minimise to 10 pages

tristyxxnana committed:
Merge branch 'populateData' into adocUpdate

zhicaizack committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T16-1/main

jomcruz93 committed:
Updated DG. Updated PPP to reduce no. of pages

harriuscai committed:
Update PPP

harriuscai committed:
Remove a section in the PPP

harriuscai committed:
Fix a slight formatting error

m-aslam-mj2 committed:
Merge pull request #349 from m-aslam-mj2/staging16

Updated student list view. Updated UG and DG. Updated build.gradle.

m-aslam-mj2 committed:
Merge pull request #351 from jomcruz93/master

Updated DG. Updated PPP to reduce no. of pages

zhicaizack committed:
* Updated DG, UG, shifted tags around
* Updated PPP

zhicaizack committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T16-1/main

tristyxxnana committed:
Merge branch 'master' into adocUpdate

m-aslam-mj2 committed:
Merge pull request #352 from zhicaizack/master

Update UG, DG, PPP

m-aslam-mj2 committed:
Merge pull request #350 from tristyxxnana/adocUpdate

Adoc update

m-aslam-mj2 committed:
Changed logo. Added tests, updated docs.

Signed-off-by: m-aslam-mj2 <e0176811@u.nus.edu>

tristyxxnana committed:
populated data

tristyxxnana committed:
Merge branch 'populate' into populatedGradebookGrade

harriuscai committed:
Add manual testing instructions

m-aslam-mj2 committed:
Merge pull request #353 from tristyxxnana/populatedGradebookGrade

Populated Gradebook Grade Data

zhicaizack committed:
* Updated PPP

m-aslam-mj2 committed:
Merge pull request #355 from zhicaizack/pppupdate

PPP Update

harriuscai committed:
Change the length of PPP

harriuscai committed:
Merge branch 'master' into master

jomcruz93 committed:
Added notes data to populate in debug command

jomcruz93 committed:
Resolved merge conflicts

m-aslam-mj2 committed:
Modified docs.

Signed-off-by: m-aslam-mj2 <e0176811@u.nus.edu>

m-aslam-mj2 committed:
Merge pull request #354 from harriuscai/master

Completed PPP and added manual testing instructions

m-aslam-mj2 committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T16-1/main

m-aslam-mj2 committed:
Fixed bug.

Signed-off-by: m-aslam-mj2 <e0176811@u.nus.edu>

m-aslam-mj2 committed:
Merge pull request #357 from jomcruz93/master

Added notes data in debug command

m-aslam-mj2 committed:
Merge pull request #356 from m-aslam-mj2/staging18

Modified docs.

m-aslam-mj2 committed:
Fix for minor error. Changed Ui.png.

Signed-off-by: m-aslam-mj2 <e0176811@u.nus.edu>

m-aslam-mj2 committed:
Merge pull request #358 from m-aslam-mj2/staging19

Fix for minor error. Changed Ui.png.
